catalog/plugin/catalog-plugin-metadatabackup/src/test/java/org/codice/ddf/catalog/plugin/metadata/backup/MetadataBackupPluginTest.java

```diff
@@ -0,0 +1,423 @@
+    }
+
+    @Test
+    public void testCreateRequestVaryingDepths() throws Exception {
+        for (int depth = 0; depth < 12; depth++) {
+            metadataBackupPlugin.setFolderDepth(depth);
+            ProcessRequest<ProcessCreateItem> newCreateRequest = metadataBackupPlugin.processCreate(
+                    createRequest);
+            assertFiles(newCreateRequest.getProcessItems(), depth);
+        }
+    }
+
+    @Test
+    public void testCreateRequestFailedTransform() throws Exception {
+        when(metacardTransformer.transform(any(Metacard.class), anyMap())).thenThrow(
+                CatalogTransformerException.class);
+        int depth = 0;
+        metadataBackupPlugin.setFolderDepth(depth);
+        metadataBackupPlugin.processCreate(createRequest);
+        assertFileDoesNotExist(depth);
+    }
+
+    @Test
+    public void testCreateRequestNullBinaryContent() throws Exception {
+        when(metacardTransformer.transform(any(Metacard.class), anyMap())).thenReturn(null);
+        int depth = 0;
+        metadataBackupPlugin.setFolderDepth(depth);
+        metadataBackupPlugin.processCreate(createRequest);
+        assertFileDoesNotExist(depth);
+    }
+
+    @Test
+    public void testCreateRequestNullBinaryContentInputStream() throws Exception {
+        BinaryContent binaryContent = mock(BinaryContent.class);
+        when(binaryContent.getInputStream()).thenReturn(null);
+        when(metacardTransformer.transform(any(Metacard.class),
+                anyMap())).thenReturn(binaryContent);
+        int depth = 0;
+        metadataBackupPlugin.setFolderDepth(depth);
+        metadataBackupPlugin.processCreate(createRequest);
+        assertFileDoesNotExist(depth);
+    }
+
+    @Test
+    public void testCreateRequestDepthGreaterThanIdLength() throws Exception {
+        int depth = 32;
+        metadataBackupPlugin.setFolderDepth(depth);
+        ProcessRequest<ProcessCreateItem> newCreateRequest = metadataBackupPlugin.processCreate(
+                createRequest);
+        assertFiles(newCreateRequest.getProcessItems(), depth);
+    }
+
+    @Test
+    public void testUpdateRequestNoDepth() throws Exception {
+        int depth = 0;
+        metadataBackupPlugin.setFolderDepth(0);
+        ProcessRequest<ProcessUpdateItem> newUpdateRequest = metadataBackupPlugin.processUpdate(
+                updateRequest);
+        assertFiles(newUpdateRequest.getProcessItems(), depth);
+    }
+
+    @Test
+    public void testUpdateRequestVaryingDepths() throws Exception {
+        for (int depth = 0; depth < 12; depth++) {
+            metadataBackupPlugin.setFolderDepth(depth);
+            ProcessRequest<ProcessUpdateItem> newUpdateRequest = metadataBackupPlugin.processUpdate(
+                    updateRequest);
+            assertFiles(newUpdateRequest.getProcessItems(), depth);
+        }
+    }
+
+    @Test
+    public void testDeleteRequestVaryingDepths() throws Exception {
+        for (int i = 0; i < 12; i++) {
```

Comment:
12 was just an arbitrary number above the MAX_FOLDER_DEPTH.  These tests are just to verify larger depths behave appropriately.  I've changed it to 5 instead (1 above MAX_FOLDER_DEPTH).